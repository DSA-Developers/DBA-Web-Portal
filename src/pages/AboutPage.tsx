import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import Container from 'react-bootstrap/Container';
import { MemberCard } from '../components/MemberCard';

import { database, storage } from '../firebase';

interface Props {}

interface CardType {
  name: string;
  photo: string;
  role: string;
  url: string;
}

const storageRef = storage.ref();
const membersRef = database.ref('members');

const useCards = () => {
  const [cardInfo, setCardInfo] = useState<Array<CardType>>([]);

  const fetchMemberInfo = async () => {
    let info: Array<CardType> = [];
    let keysNoURL: Array<number> = [];

    const snapshot = await membersRef.get();
    snapshot.forEach((member: any) => {
      if (!member.val().url) {
        keysNoURL.push(member.key);
      }
      let memberInfo: CardType = {
        name: member.val().name,
        role: member.val().role,
        photo: member.val().photo,
        url: '' || member.val().url,
      };
      info = [...info, memberInfo];
    });

    if (keysNoURL.length !== 0) {
      await fetchPhotos(info, keysNoURL);
    }

    setCardInfo(info);
  };

  const fetchPhotos = async (info: Array<CardType>, keys: Array<number>) => {
    for (const index of keys) {
      const member = info[index - 1];
      const url = await storageRef.child('members/' + member.photo).getDownloadURL();
      info[index - 1] = { ...member, url: url };
      membersRef.child(`${index}`).set({
        name: member.name,
        role: member.role,
        photo: member.photo,
        url: url,
      });
    }
    return info;
  };

  useEffect(() => {
    fetchMemberInfo();
  }, []);

  return cardInfo;
};

export const AboutPage: React.FC<Props> = ({}) => {
  const cardInfo = useCards();

  const cards = cardInfo.map((card) => (
    <MemberCard key={card.url} name={card.name} position={card.role} fileName={card.url} />
  ));

  return (
    <Container fluid>
      <DescriptionContainer>
        <Title>About us</Title>
        <Description>
          The Dominican Student Association was first established at the University of Florida in 2000. We are a
          welcoming club for all those interested in the Dominican Republic and who wish to make a difference in the
          lives of not only Dominican people but as many people as possible. DSA strives to create a community of
          Dominicans at the university. DSA hosts monthly General Body Meetings and social events, and during the Spring
          semester, a full week is dedicated to celebrating Dominican Independence. DSA also partakes in drives such as
          the holiday toy drive for kids in Haina, Dominican Republic and dance workshops with GatorSalsa.
        </Description>
      </DescriptionContainer>
      <MemberContainer>
        <Title>Executive Board</Title>
        <ParentAffiliation>{cards.slice(0, 2)}</ParentAffiliation>
        <Affiliations2>{cards.slice(2, 5)}</Affiliations2>
        <Affiliations2>{cards.slice(5)}</Affiliations2>
      </MemberContainer>
    </Container>
  );
};

const DescriptionContainer = styled.div`
  padding: 20px 400px 50px 400px;
  background-color: #f8f9fb;
  textalign: 'center';
`;
const MemberContainer = styled.div`
  background-color: #fff;
  margin: -30px 0px 60px 0px;
`;

const Description = styled.h1`
  font-family: Optima, Segoe, 'Segoe UI', Candara, Calibri, Arial, sans-serif;
  font-size: 20px;
  font-style: normal;
  font-variant: normal;
  font-weight: 350;
  line-height: 40px;
  padding: 2% 3%;
  text-align: center;
`;

const Title = styled.div`
  font-family: Optima, Segoe, 'Segoe UI', Candara, Calibri, Arial, sans-serif;
  font-size: 50px;
  font-style: normal;
  font-variant: small-caps;
  font-weight: 700;
  padding: 4%;
  margin-bottom: 1%;
  text-align: center;
  background: -webkit-linear-gradient(0.5turn, #ce1126, #002d62);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
`;

const ParentAffiliation = styled.div`
  margin: 0 0 100px 0;
  display: grid;
  grid-template-columns: repeat(2, 200px);
  column-gap: 100px;
  justify-content: center;
`;

const Affiliations2 = styled(ParentAffiliation)`
  grid-template-columns: repeat(3, 200px);
`;
