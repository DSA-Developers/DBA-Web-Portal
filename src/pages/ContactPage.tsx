import React from 'react';

interface Props {}

export const ContactPage: React.FC<Props> = ({}) => {
  return (
    <div>
      <h1>Contact Page</h1>
    </div>
  );
};
