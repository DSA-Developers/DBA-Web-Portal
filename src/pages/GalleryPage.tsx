import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import Container from 'react-bootstrap/Container';
import { CenterRow } from '../components/CenterRow';
import { Title } from '../components/Title';
import { storage } from '../firebase';
import Gallery from 'react-photo-gallery';
import { SRLWrapper } from 'simple-react-lightbox';

interface Props {}

const photosRef = storage.ref().child('photos');

const usePhotos = () => {
  const [photos, setPhotos] = useState<Array<string>>([]);

  const fetchPhotos = async () => {
    let requests: Array<Promise<any>> = [];
    const list = await photosRef.listAll();
    list.items.forEach((itemRef) => {
      requests.push(itemRef.getDownloadURL());
    });
    const urls: Array<string> = await Promise.all(requests);
    console.log(urls);
    setPhotos(urls);
  };

  useEffect(() => {
    fetchPhotos();
  }, []);

  return photos;
};

export const GalleryPage: React.FC<Props> = ({}) => {
  const photosURLs = usePhotos();

  const photos = photosURLs.map((url) => ({ src: url, height: 7, width: 10 }));

  return (
    <>
      <StyledContainer fluid>
        <CenterRow>
          <Title>Photo Gallery</Title>
          <SRLWrapper>
            <Gallery photos={photos} />
          </SRLWrapper>
          <Title></Title>
        </CenterRow>
      </StyledContainer>
    </>
  );
};

const StyledContainer = styled(Container)`
  background-color: #f8f9fb;
  padding: 0 10% 0 10%;
`;
