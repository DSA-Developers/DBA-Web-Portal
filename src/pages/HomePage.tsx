import React from 'react';
import styled from 'styled-components';
import Container from 'react-bootstrap/Container';
import { CenterRow } from '../components/CenterRow';
import Button from 'react-bootstrap/Button';

interface Props {}

const Title = styled.h1`
  font-family: Roboto;
`;

export const HomePage: React.FC<Props> = ({}) => {
  return (
    <Container>
      <CenterRow>
        <Title>Home Page</Title>
      </CenterRow>
      <CenterRow>
        <h2>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi fermentum elementum lorem. Quisque ac dapibus lacus, in sagittis leo. Praesent fermentum nec massa et imperdiet. Fusce sapien nisl, lacinia ut ante eu, laoreet elementum ex. Duis eget purus a dui pellentesque maximus. Aenean vel ultricies velit. Mauris scelerisque sapien dolor, a laoreet magna aliquet ut. Vestibulum nec dignissim tellus.
        Recusandae autem et ipsam. Saepe qui et repellat. Maxime distinctio non occaecati cum voluptates voluptatibus quo minima. Inventore facilis a accusantium quaerat ut quam quos. Enim qui porro nobis in qui ratione.
        Porro non maxime et porro veniam ut laudantium. Magni quidem non officiis eaque nobis quidem. Laboriosam doloremque quos eius. Maiores quia est explicabo nisi consectetur. Consectetur velit ducimus maiores iusto quia natus aliquid. Odio libero magnam praesentium accusantium pariatur quia perspiciatis incidunt.
        Provident consectetur quo nostrum veniam. Nam inventore ullam repellat soluta corrupti quam. Dolor quis aliquam exercitationem rerum iure placeat voluptatem.
        Id non rem reprehenderit dolor similique et. Et ut laborum velit incidunt. Ea excepturi et repudiandae. Rem culpa rerum dolores accusantium sed magnam facilis officia.
        Quaerat exercitationem minima dolor amet rerum nesciunt est occaecati. Labore quidem quae excepturi. Et praesentium quidem nihil quos.
        </h2>
      </CenterRow>
    </Container>
  );
};
