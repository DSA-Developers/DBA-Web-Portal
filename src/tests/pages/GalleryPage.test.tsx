import renderer from 'react-test-renderer';
import { GalleryPage } from '../../pages/GalleryPage';

test('Gallery Page renders', () => {
  const props = {};
  const component = renderer.create(<GalleryPage {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});