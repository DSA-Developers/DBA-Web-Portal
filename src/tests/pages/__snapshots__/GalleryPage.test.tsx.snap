// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Page renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-family: Optima,Segoe,'Segoe UI',Candara,Calibri,Arial,sans-serif;
  font-size: 50px;
  font-style: normal;
  font-variant: small-caps;
  font-weight: 700;
  padding: 4%;
  margin-bottom: -2%;
  text-align: center;
  background: -webkit-linear-gradient(0.5turn,#ce1126,#002d62);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

<div
  className="container"
>
  <div
    className="c0 row"
  >
    <div
      className="c1"
    >
      Photo Gallery Page
    </div>
  </div>
</div>
`;
